test-utils/src/bitcoin_manager.rs

```diff
@@ -28,54 +26,77 @@ impl<'a> BitcoinManager<'a> {
-            manager,
-            p2p_port,
-            rpc_port,
-            network: settings.bitcoin_network.to_string(),
-            settings: settings.clone(),
-            client: OnceLock::new(),
-        };
-        settings.bitcoind_rpc_port = bitcoind.rpc_port;
-        settings.bitcoin_cookie_path = bitcoind.cookie_path();
-        bitcoind.settings.bitcoind_rpc_port = settings.bitcoind_rpc_port;
-        bitcoind.settings.bitcoin_cookie_path = settings.bitcoin_cookie_path.clone();
-        bitcoind
-            .start(BitcoindCheck(bitcoind.settings.clone()))
-            .await?;
-        Ok(bitcoind)
-    }
+        settings.bitcoind_rpc_port = rpc_port;
+        settings.bitcoin_cookie_path = if settings.bitcoin_network == Network::Main {
+            storage_dir.join(".cookie")
+        } else {
+            storage_dir
+                .join(settings.bitcoin_network.to_string())
+                .join(".cookie")
+        }
+        .into_os_string()
+        .into_string()
+        .expect("should not use non UTF-8 char in path");
 
-    pub async fn start(&mut self, check: impl Check) -> Result<()> {
         let args = &[
             "-server",
             "-noconnect",
-            "-rpcthreads=16",
+            "-rpcthreads=1",
             "-listen",
-            &format!("-chain={}", &self.network),
-            &format!("-datadir={}", &self.manager.storage_dir.as_path().display()),
-            &format!("-port={}", &self.p2p_port.to_string()),
-            &format!("-rpcport={}", &self.rpc_port.to_string()),
+            &format!("-chain={}", settings.bitcoin_network),
+            &format!("-datadir={}", storage_dir.display()),
+            &format!("-port={}", p2p_port),
+            &format!("-rpcport={}", rpc_port),
+            // NOTE
+            // Uncomment it for debugging , there is not good reason always log
+            // Will integrate with #619
+            // &format!("-debuglogfile=<file>", storage_dir.join("bitcoind.log").display())
         ];
-        self.manager.start("bitcoind", args, check).await?;
-        self.client
-            .set(BitcoindClient::new(&self.settings).await?)
-            .unwrap_or_default();
-        Ok(())
-    }
 
-    pub fn cookie_path(&self) -> String {
-        let dir = if self.network == "mainnet" {
-            self.manager.storage_dir.clone()
-        } else {
-            self.manager.storage_dir.join(&self.network)
+        let mut process = Command::new("bitcoind")
+            .args(args)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+            .with_context(|| "failed to start bitcoind")?;
+
+        // XXX
+        // Request `call` and `new` of BitcoindClient should be separate functions
+        // It is not `BitcoindConnection`, so it doesn't make sense to have a hidden request in new.
+        // Also the manager is hard to init the client with `bitcoind` at same time.
```

Comment:
Please don't leave this comment, create an issue if it can't be fixed now.

Reply:
Why should I open an issue before a code merge? And how can I pin the code snip?